examples/lua/Makefile

```diff
@@ -0,0 +1,53 @@
+APPLICATION = lua
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= native
+
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/../..
+
+# Comment this out to disable code in RIOT that does safety checking
+# which is not needed in a production environment but helps in the
+# development process:
+DEVELHELP ?= 1
+
+# Change this to 0 show compiler invocation lines by default:
+QUIET ?= 1
+
+USEMODULE += ps
+USEMODULE += shell
+USEMODULE += shell_commands
+
+ifneq ($(BOARD),native)
+# This stack size is large enough to run Lua print() functions of
+# various lengths. Other functions untested.
+CFLAGS += -DTHREAD_STACKSIZE_MAIN=4096
+endif
+
+USEPKG += lua
+
+include $(RIOTBASE)/Makefile.include
+
+# The code below generates a header file from any .lua scripts in the
+# example directory. The header file contains a byte array of the
+# ASCII characters in the .lua script.
+
+LUA_PATH := $(BINDIR)/lua
```

Comment:
This lines can be moved to the pkg Makefiles and invoked when USEPKG+=lua. See e.g [here](https://github.com/RIOT-OS/RIOT/blob/master/pkg/openthread/Makefile#L16)

Reply:
on second thoughts, maybe it's not bad to leave it as is. It explicitly tells the developer where's the file. But it could also be documented in the README.md. I'm OK with both ;)

Reply:
I guess this wasn't too clear - this path variable is only used in this example, to contain the path where the script should be parsed to. I've renamed this LUA_SCRIPT_HEADER_PATH to avoid confusion and documented it in the readme. Also, it now builds to examples/lua/scripts. What do you think?

Reply:
I'd prefer to keep it as it is, for consistency with examples/javascript and pkg/jerryscript which do the same thing.

Reply:
ACK

Reply:
maybe we can think about this later

Reply:
Where's the best place for this kind of discussion after a PR has been closed?

Reply:
Another PR/RFC should do the trick